.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,164 @@
+name: Pull Request
+
+permissions:
+  contents: read
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled]
+  workflow_dispatch:
+  workflow_call:
+    inputs:
+      publish_dev_version:
+        description: "Publish the generated dev version to riv-pypi"
+        default: false
+        required: false
+        type: boolean
+
+jobs:
+  # Define the matrix of deltakit subprojects that have been modified in the PR and
+  # should be tested. This is done by looking at tags, so label_pr.yml should have done
+  # its job before.
+  # Because there is no easy way to ensure that label_pr.yml runs before this workflow,
+  # this job returns all the packages when no tags are set.
+  define-project-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      projects: ${{ steps.json_labels.outputs.projects }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v6
+      - name: Set up Python
+        uses: actions/setup-python@v6
+        with:
+          python-version: "3.10"
+      - name: Outputs modified packages that needs to be tested.
+        id: json_labels
+        run: python tools/transform_labels.py
+        env:
+          projects: ${{ join(github.event.pull_request.labels.*.name, ',') }}
+
+  # Build deltakit wheels.
+  build:
+    runs-on: ubuntu-latest
+    needs: define-project-matrix
+    strategy:
+      matrix:
+        projects: ${{ fromJSON(needs.define-project-matrix.outputs.projects) }}
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Install uv
+        uses: astral-sh/setup-uv@v7
+
+      - name: Build deltakit
+        run: uv build --no-sources --package ${{ matrix.projects.project }}
+
+      - name: Upload ${{ matrix.projects.project }} wheels
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.projects.project }}-wheel
+          path: ./dist/*.whl
+          retention-days: 7
```

Comment:
Is that a correct time? It seems like 10 minutes (or 1 hour to be extra large) would be enough, because we upload the packages to the private PyPI repository at the end of CI.